literal/decimal_util.go

```diff
@@ -0,0 +1,73 @@
+package literal
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+)
+
+func decimalStringToBytes(decimalStr string) ([16]byte, int32, int32, error) {
+	var result [16]byte
```

Comment:
does it makes since to use a utility library such as https://github.com/cockroachdb/apd avoid hand coding this?

Reply:
We already include "math/big", we can use big.ParseFloat, or we can use the Go Arrow library decimal128 package which has a FromString method, etc. Personally I'd prefer those

Reply:
Another example that can be used is https://github.com/zeroshade/iceberg-go/blob/main/literals.go#L996